### PR TITLE
Add support for Filestream input in elastic agent.

### PR DESCRIPTION
**Pull request by ph**
Add support for the filestream input and regenerate the spec.
```
inputs:
 - type: filestream
 ```

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: x-pack/elastic-agent/spec/filebeat.yml [hunk 1]
```diff
@@ -80,6 +80,7 @@ rules:
     - udp
     - unix
     - winlog
+    - filestream
```
Comment (urso) [inline x-pack/elastic-agent/spec/filebeat.yml:83]: Do we want to expose it as filestream, or `event/filestream` via the mapper?

Comment (ph) [reply x-pack/elastic-agent/spec/filebeat.yml:83]: Good point, I don't have a preference, `event/filestream` seems like a good idea? @mostlyjason @ruflin ?

Comment (ruflin) [reply x-pack/elastic-agent/spec/filebeat.yml:83]: No strong preference on my end. Main downside of the prefix is that it is more cumbersome to type and harder to remember?

Comment (mostlyjason) [reply x-pack/elastic-agent/spec/filebeat.yml:83]: No preference for me

Comment (urso) [reply x-pack/elastic-agent/spec/filebeat.yml:83]: I would like us to maintain some kind of consistency for input names that are available via Agent. Unfortunately Filebeat input naming is not really consistent with the other Beats and even within Filebeat. As the input is not GA input I'm okay to keep it named `filestream`.

We already have `event/log` and the `logfile` inputs, each translating to `logs`. We will eventually replace `logs` with `filestream`. The `filestream` input is not 100% bc, that is we will also have to think about migration.

Comment (ph) [reply x-pack/elastic-agent/spec/filebeat.yml:83]: Agree with the proposal @urso, can you give a review on this PR?

**Review by ruflin — APPROVED** [review]

**Review by urso — APPROVED** [review]

**Pull request merged**